scripts/dev-environment.sh

```
@@ -506,29 +506,77 @@ check_balances() {
+            print_status "Stopping Core Lane log viewer (found by pattern)..."
+            echo "$core_tail_pids" | xargs kill 2>/dev/null || true
+        fi
     fi
 
-    if [ $MINING_PID -ne 0 ]; then
+    # Stop mining loop - try PID first, then find by pattern
+    if [ $MINING_PID -ne 0 ] && kill -0 $MINING_PID 2>/dev/null; then
         print_status "Stopping mining loop (PID: $MINING_PID)..."
         kill $MINING_PID 2>/dev/null || true
+    else
+        local mining_pids=$(ps aux | grep -E "bash.*sleep 10|bash.*while true" | grep -v "grep\|status\|stop\|cleanup" | awk '{print $2}' 2>/dev/null || true)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)